panel/viewable.py

```diff
@@ -631,6 +633,18 @@ def __repr__(self, depth: int = 0) -> str:
         return '{cls}({params})'.format(cls=type(self).__name__,
                                         params=', '.join(param_reprs(self)))
 
+    @property
+    def component_name(self) -> str:
```

Comment:
Would definitely much prefer not adding this. Can you explain what was wrong with simply using the `.name` parameter value?

Reply:
I totally agree that `.name` can be used but it has a problem - For some components `name` appears in the UI e.g. button. Now consider a page that has two forms. Each form has a button with name `Submit` in the UI. Now, when user clicks one of the button there is no way to know from the logs (logged using this middleware feature) which button was clicked. This property gives developers an option to add custom business specific name that can be used for logging purposes, e.g. two buttons in same page can be `pn.widgets.Button(name='Submit', component_name='SUBMIT_FORM_X')` and `pn.widgets.Button(name='Submit', component_name='SUBMIT_FORM_Y')`.

Button is just one example, idea was to provide users a way so they can choose different name for what appears in UI v/s what is logged and both can be different. Let me know your thoughts.

Reply:
Yeah, it was an unfortunate mistake of mine in the early design of Panel to use the `name` as the title/label for various widgets.

Reply:
To make it less confusing with `name`, we can name this property to `logging_name`, `loggable_name`, `component_logging_name`, `log_tag`, etc.

Reply:
@philippjfr Given above, should I completely remove `component_name` for now, if we do not plan to re-name it to say `loggable_name` or something? 

I'll note another problem with `name` is that by default each widget takes arbitrary name which are not meaningful - https://github.com/holoviz/panel/issues/1294#issuecomment-619232730

Reply:
I agree with you @nishikantparmariam and I share the same use case with multiple Button widgets having `'Submit'` for their `name` value. But I also share the same concerns of Philipp adding a new argument to the constructor. @philippjfr how about abusing `tags` with e.g. `Button(name='Submit', tags=[{'debug_name': 'MyComponent'}])`?

Reply:
> how about abusing tags with e.g. `Button(name='Submit', tags=[{'debug_name': 'MyComponent'}])`?

To me at least, it seems great! I wasn't aware of `tags`, as it is a recent addition.

Reply:
Don't we have a distinction between `name` and `label`, where the name is the _default_ label, but an explicit label can override it? Seems like that could apply here, just as it does in other HoloViz contexts.

Reply:
That seems like a nice way to begin migrating away from using `name` as the parameter controlling widget titles.

Reply:
Removed `component_name`, I'll use `tags` for my use-case.